### PR TITLE
Table: Add dataframe name to single table

### PR DESCRIPTION
**Pull request by joey-grafana**
**What is this feature?**

Update table name if only one table i.e. one query.

**Why do we need this feature?**

Adds context to the table itself. Myself and Jen were speaking about this and think it would be a good addition for the metrics summary table and could benefit other users too.

**Who is this feature for?**

Users of explore / table.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/Table/TableContainer.tsx [hunk 1]
```diff
@@ -88,7 +88,13 @@ export class TableContainer extends PureComponent<Props> {
...
             <PanelChrome
               key={data.refId || `table-${i}`}
-              title={dataFrames && dataFrames.length > 1 ? `Table - ${data.name || data.refId || i}` : 'Table'}
+              title={
+                dataFrames && dataFrames.length > 1
+                  ? `Table - ${data.name || data.refId || i}`
+                  : data.name
+                  ? `Table - ${data.name}`
+                  : 'Table'
+              }
```
Comment (Elfo404) [inline public/app/features/explore/Table/TableContainer.tsx:97]: this is getting quite hard to follow, can we move it somewhere?

Comment (ifrost) [reply public/app/features/explore/Table/TableContainer.tsx:97]: Yeah, with a bit more code it will be more readable:

```typescript

let tableTitle;
if (dataFrames && dataFrames.length > 1) {
  tableTitle = `Table - ${data.name || data.refId || i}`;
} else {
  tableTitle = data.name ? `Table - ${data.name}` : "Table";
}
```

**Review by ifrost — APPROVED** [review]
